services/catalog/rs_server_catalog/user_catalog.py

```diff
@@ -1141,65 +1207,121 @@ async def retrieve_timestamp(self, request: Request) -> tuple[str, str]:
+        # Dictionary to easily access main data from the request
+        self.request_ids = {
+            "auth_roles": auth_roles,
+            "user_login": user_login,
+            "owner_id": owner_id,
+            "collection_id": [],
+            "item_id": "",
+        }
+        reroute_url(request, self.request_ids)
         if not request.scope["path"]:  # Invalid endpoint
             return JSONResponse(content="Invalid endpoint.", status_code=HTTP_400_BAD_REQUEST)
         logger.debug(f"reroute_url formating: path = {request.scope['path']} | requests_ids = {self.request_ids}")
-        # Overwrite user and collection id with the ones provided in the request body
+
+        # Ensure that user_login is not null after rerouting
+        if not self.request_ids["user_login"]:
+            raise HTTPException(
+                status_code=500,
+                detail="user_login is not defined !",
+            )
+
+        # ---------- Body data recovery
+        # Recover user and collection id with the ones provided in the request body
+        # (if the corresponding parameters have not been recovered from the url)
         # This is available in POST/PUT/PATCH methods only
         if request_body:
-            self.request_ids["owner_id"] = self.request_ids.get("owner_id") or request_body.get("owner")
+            # Edit owner_id with the corresponding body content if exist
+            if not self.request_ids["owner_id"]:
+                self.request_ids["owner_id"] = request_body.get("owner")
             # received a POST/PUT/PATCH for a STAC item or
             # a STAC collection is created
-            if not self.request_ids["collection_id"]:
-                self.request_ids["collection_id"] = request_body.get("collection") or request_body.get("id")
+            if len(self.request_ids["collection_id"]) == 0:
+                collections = request_body.get("collections") or request_body.get("id")
+                if collections:
+                    self.request_ids["collection_id"] = collections if isinstance(collections, list) else [collections]
+
             if not self.request_ids["item_id"] and request_body.get("type") == "Feature":
                 self.request_ids["item_id"] = request_body.get("id", None)
 
         if "/health" in request.scope["path"]:
             # return true if up and running
             return JSONResponse(content="Healthy", status_code=HTTP_200_OK)
-        # Handle requests
-        if request.scope["path"] == "/search":
-            # URL: GET: '/catalog/search'
-            request = await self.manage_search_request(request)
-            if hasattr(request, "status_code"):  # Unauthorized
-                return request
-        elif request.method in {"POST", "PUT"} and self.request_ids["owner_id"]:
+
+        # ---------- Apply specific changes for each endpoint
+
+        if request.method in {"POST", "PUT"} and "/search" not in request.scope["path"]:
```

Comment:
Not very important, but fyi, it's optimal to use this operator over Tuples, not sets, since are immutables thus lower memory footprint